js/app/modules/contentGroup/contentGroup.js

```diff
@@ -166,9 +166,14 @@ const ContentGroup = new Module.Class({
 
     _on_models_changed: function () {
         let models = this._selection.get_models();
-        if (models.length === 0 && this._no_results) {
+        this.visible = true;
+        if (models.length === 0) {
             this._arrangement.set_models([]);
-            this._stack.visible_child_name = NO_RESULTS_PAGE_NAME;
+            if (this._no_results) {
```

Comment:
OK, I got confused with the mention of search results. I understand now.

We were already doing this in news.yaml with the set page; how come it worked properly there?